parsl/dataflow/dflow.py

```diff
@@ -367,6 +367,12 @@ def launch_if_ready(self, task_id):
                             task_id, self.tasks[task_id]['func'], *new_args, **kwargs)
 
                 if exec_fu:
+
+                    try:
+                        exec_fu.add_done_callback(partial(self.handle_exec_update, task_id))
```

Comment:
We've had  issues with the order in which callbacks are added creating race-conditions/ breakages. Would be useful to run this through more rounds of testing than usual.